UIforETW/Utility.cpp

```diff
@@ -431,108 +645,117 @@ bool Is64BitWindows()
 #else
 	// http://blogs.msdn.com/b/oldnewthing/archive/2005/02/01/364563.aspx
 	BOOL f64 = FALSE;
-	bool bIsWin64 = IsWow64Process(GetCurrentProcess(), &f64) && f64;
-	return bIsWin64;
+	return ::IsWow64Process(::GetCurrentProcess(), &f64) && f64;
 #endif
 }
 
 bool Is64BitBuild()
 {
+	//when MSVC gets constexpr, this is a perfect candidate.
 	return sizeof(void*) == 8;
 }
 
-WindowsVersion GetWindowsVersion()
+bool IsWindowsTenOrGreater()
 {
-	OSVERSIONINFO verInfo = { sizeof(OSVERSIONINFO) };
-	// warning C4996: 'GetVersionExA': was declared deprecated
-	// warning C28159: Consider using 'IsWindows*' instead of 'GetVersionExW'.Reason : Deprecated.Use VerifyVersionInfo* or IsWindows* macros from VersionHelpers.
-#pragma warning(suppress : 4996)
-#pragma warning(suppress : 28159)
-	GetVersionEx(&verInfo);
-
-	// Windows 10 preview has major version 10, if you have a compatibility
-	// manifest for that OS, which this program should have.
-	if (verInfo.dwMajorVersion > 6)
-		return kWindowsVersion10;	// Or higher, I guess.
-
-	// Windows 8.1 will only be returned if there is an appropriate
-	// compatibility manifest.
-	if (verInfo.dwMajorVersion == 6 && verInfo.dwMinorVersion >= 3)
-		return kWindowsVersion8_1;
-
-	if (verInfo.dwMajorVersion == 6 && verInfo.dwMinorVersion >= 2)
-		return kWindowsVersion8;
+	return IsWindowsVersionOrGreater(10, 0, 0);
+}
 
-	if (verInfo.dwMajorVersion == 6 && verInfo.dwMinorVersion >= 1)
-		return kWindowsVersion7;
+bool IsWindowsXPOrLesser()
+{
+	return !IsWindowsVistaOrGreater();
+}
 
-	if (verInfo.dwMajorVersion == 6)
-		return kWindowsVersionVista;
+bool IsWindowsSevenOrLesser()
+{
+	return !IsWindows8OrGreater();
+}
 
-	return kWindowsVersionXP;
+bool IsWindowsVistaOrLesser()
+{
+	return !IsWindows7OrGreater();
 }
 
+std::wstring GetEnvironmentVariableString(_In_z_ PCWSTR const variable)
+{
+	const rsize_t bufferSize = 512u;

```

Comment:
I agree, this should be fixed. Not sure why everything ran fine on my machine? My PATH is definitely more than 512 characters. I've had trouble with NSIS installers because of it.